contracts/contracts/src/functioncall/gpact/CrosschainControl.sol

```diff
@@ -452,6 +454,8 @@ contract CrosschainControl is
                 );
                 activeCallFailed = true;
             }
+        } else {
+            revert("Cross Blockchain Call failed");
```

Comment:
btw, a reason why throwing a revert doesn't work is that a contract in the middle of the call could catch the revert and ignore it. That is: CrosschainControl.sol:makeCall -> Application Contract1.sol:some function -> Application Contract2.sol:some function -> CrosschainControl.sol:crossBlockchainCall. The Application Contract2 could catch the revert and then ignore it. This would result in the failure not being seen by CrosschainControl.sol:makeCall.

Reply:
I tried to remove this line and the integration test fails. Check [this](https://app.circleci.com/pipelines/github/ConsenSys/gpact/710/workflows/a86a0762-9129-49d6-8329-aee47ef42687) (without this revert) and [this] (https://app.circleci.com/pipelines/github/ConsenSys/gpact/711/workflows/e95285f7-8382-4911-9d5d-142b5c981c0b) (with revert).  

The reason is the root call doesn’t know whether the call failed or not.   

The failure testing scenario is that user wish to purchase an NFT on chain A using fund on chain B - Initially the NFT was listed for a price of 100 tokens, but during the process of segment call, seller increases the NFT to be a price of 110 tokens.  GPACT flow should be like -> start transaction on chain A (start), provisional token transfer of 100 on chain B (segment, succeed) -> root transaction on chain A (should failed, as root transaction will use a crosschainCall with parameters of current price 110, and there is no return value for this call).  However, if you take a look at [here](https://github.com/ConsenSys/gpact/blob/4f83c064c3e459615198b33383f73f8b310388bb/contracts/contracts/src/functioncall/gpact/CrosschainControl.sol#L351), root uses makecall() to see if the call of this root transaction succeed or not, makecall() further calls [here](https://github.com/ConsenSys/gpact/blob/4f83c064c3e459615198b33383f73f8b310388bb/contracts/contracts/src/functioncall/gpact/CrosschainControl.sol#L551) targetContract.call() which further calls [here](https://github.com/ConsenSys/gpact/blob/4f83c064c3e459615198b33383f73f8b310388bb/contracts/contracts/src/functioncall/gpact/CrosschainControl.sol#L435) to transfer 110 tokens on remote chain. Although the call is failed, cross blockchain call does not throw any error. From application logic, the call succeed => therefore targetContract.call() succeed => makecall() succeed => root call succeed, and commit message goes to chain B rather than a discard.